src/components/select-project/select-project.controller.ts

```diff
@@ -161,13 +161,14 @@ export class SelectProjectController implements angular.IController {
   private canIAddToProject(): boolean {
     let canIAddToProject: boolean = true;
 
-    // TODO AuthorizationService.canIAddToProject assumes that the ProjectsService.get has already completed performed
-    // but currently this happens after project change events fire and it happens in the various dialog controllers, not
-    // as part of this component.  For now always return true until we can resolve this.
-    //
-    // if (!this.isNewProject()) {
-    //   canIAddToProject = this.AuthorizationService.canIAddToProject(_.get(this.ctrl.selectedProject, 'metadata.name'));
-    // }
+    var projectName = _.get(this.ctrl.selectedProject, 'metadata.name');
+
+    if (!this.isNewProject()) {
+      this.AuthorizationService.getProjectRules(projectName).then( () => {
+        canIAddToProject = this.AuthorizationService.canIAddToProject(projectName);
+        return this.ctrl.forms.selectProjectForm.selectProject.$setValidity('cannotAddToProject', canIAddToProject);
```

Comment:
This value is not actually returned to anyone. If you need `canIAddToProject()` to always return the result, it needs to return a promise that's resolved. I don't see anyone actually using the return value from `canIAddToProject()`, however, so I'd just remove the return here and below.

Note that this line also changes the return value for anyone else sharing this promise from AuthorizationService!

Reply:
Ok, done, thanks!